docs/source/app_developers_guide/zmq_event_subscription.rst

```diff
@@ -0,0 +1,309 @@
+   messages must be compiled for the chosen language.
+
+The following steps assume that the XO transaction family has a ``create`` event
+that is sent when a game has been created, as in this example:
+
+.. code-block:: python
+
+     context.add_event(
+         "xo/create", {
+             'name': name,
+             'creator', signer_public_key
+     })
+
+
+Step 1: Construct a Subscription
+================================
+
+An application can use the ``EventSubscription`` protobuf message to construct
+an event subscription. For example, in the ``sawtooth`` namespace, the
+application could subscribe to either a ``block-commit`` or ``state-delta``
+event (or both) in the ``sawtooth`` namespace, using either a ``SIMPLE`` or
+``REGEX`` filter.
+
+The following example constructs an event subscription for state-delta events
+(changes in state) with a ``REGEX_ANY`` filter for events from the XO
+transaction family.
+
+.. code-block:: python
+
+    subscription = EventSubscription(
+        event_type="sawtooth/state-delta",
+        filters=[
+            # Filter to only addresses in the "xo" namespace using a regex
+            EventFilter(
+                key="address",
+                match_string="xo.*",
+                filter_type=EventFilter.REGEX_ANY)
+        ])
+
+
+Step 2: Submit the Event Subscription
+=====================================
+
+After constructing a subscription, submit the subscription request to the
+validator. The following example connects to the validator using ZMQ, then
+submits the subscription request.
+
+.. code-block:: python
+
+    # Setup a connection to the validator
+    ctx = zmq.Context()
+    socket = ctx.socket(zmq.DEALER)
+    socket.connect(url)
+
+    # Construct the request
+    request = ClientEventsSubscribeRequest(
+        subscriptions=[subscription]).SerializeToString()
+
+    # Construct the message wrapper
+    correlation_id = "123" # This must be unique for all in-flight requests
+    msg = Message(
+        correlation_id=correlation_id,
+        message_type=CLIENT_EVENTS_SUBSCRIBE_REQUEST,
+        content=request)
+
+    # Send the request
+    socket.send_multipart([msg.SerializeToString()])
+
+.. note::
+
+   For information about ``ClientEventsSubscribeRequest``, see
+   :doc:`about_event_subscriptions`.
+
+
+Step 3: Receiving the Response
+==============================
+
+After submitting the subscription request, wait for a response from the
+validator. The validator will return a response indicating whether the
+subscription was successful.
+
+The following example receives the response and verifies the status.
+
+.. code-block:: python
+
+    # Receive the response
+    resp = socket.recv_multipart()[-1]
+
+    # Parse the message wrapper
+    msg = Message()
+    msg.ParseFromString(resp)
+
+    # Validate the response type
+    if msg.message_type != CLIENT_EVENTS_SUBSCRIBE_RESPONSE:
+        print("Unexpected message type")
```

Comment:
This should probably also have a `return`

Reply:
To a tech writer, "return" means a carriage return (aka new line or line break). I'm guessing that's not what you mean.  Could you supply an example "return" for this so-not-a-Python-programmer?  Thanks!

Reply:
Lol, it's literally just a return. Sorry for the confusion.
+    if msg.message_type != CLIENT_EVENTS_SUBSCRIBE_RESPONSE:
+        print("Unexpected message type")
+        return

Reply:
Done